lib/plugins/config/admin.php

```diff
@@ -105,53 +107,68 @@ public function html()
+                    echo '</section>';
                 } else {
                     $in_fieldset = true;
                 }
                 if ($first_plugin_fieldset && $setting->getType() == 'plugin') {
-                    $this->printH1('plugin_settings', $this->getLang('_header_plugin'));
+                    $this->printHeadline(kHeadlineLevel, 'plugin_settings', $this->getLang('_header_plugin'));
                     $first_plugin_fieldset = false;
                 } elseif ($first_template_fieldset && $setting->getType() == 'template') {
-                    $this->printH1('template_settings', $this->getLang('_header_template'));
+                    $this->printHeadline(kHeadlineLevel, 'template_settings', $this->getLang('_header_template'));
                     $first_template_fieldset = false;
                 }
-                echo '<fieldset id="' . $setting->getKey() . '">';
-                echo '<legend>' . $setting->prompt($this) . '</legend>';
-                echo '<div class="table">';
-                echo '<table class="inline">';
+                echo '<section id="' . $setting->getKey() . '">';
+                echo '<h3>' . $setting->prompt($this) . '</h3>';
+                echo '<div class="settings">';
+
             } else {
                 // config settings
                 [$label, $input] = $setting->html($this, $this->hasErrors);
 
-                $class = $setting->isDefault()
-                    ? ' class="default"'
-                    : ($setting->isProtected() ? ' class="protected"' : '');
-                $error = $setting->hasError()
-                    ? ' class="value error"'
-                    : ' class="value"';
-                $icon = $setting->caution()
-                    ? '<img src="' . self::IMGDIR . $setting->caution() . '.png" ' .
-                    'alt="' . $setting->caution() . '" title="' . $this->getLang($setting->caution()) . '" />'
-                    : '';
-
-                echo '<tr' . $class . '>';
-                echo '<td class="label">';
-                echo '<span class="outkey">' . $setting->getPrettyKey() . '</span>';
-                echo $icon . $label;
-                echo '</td>';
-                echo '<td' . $error . '>' . $input . '</td>';
-                echo '</tr>';
+                // build the classlist and status text:
+                $classlist = [];
+				$status = [];
+				array_push($status, ($setting->isDefault() ? 'default' : 'modified'));
+                if ($setting->isDefault()) array_push($classlist, 'default');
+                if ($setting->isProtected()) {
+					array_push($classlist, 'protected');
+					array_push($status, 'protected');
+				}
+                if ($setting->hasError()) {
+					array_push($classlist, 'error');
+					array_push($status, 'error');
+				}
+                if ($setting->caution()) array_push($classlist, 'caution');
+
+				$makeLi = function(string $it): string {
```

Comment:
Why redeclare this method in every loop? Make it a class method.

Reply:
I went for a closure instead. Apparently the compiler optimizes these into a separate "Closure" class, so the performance impact should be manageable, and that also avoids polluting the class scope with single-use private methods.